### PR TITLE
FIX-#4924: fix read_excel when header is None

### LINKED ISSUE #4924: read_excel function fail when header is set to None
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: MacOS
- **Modin version** (`modin.__version__`): 0.5.12
- **Python version**: 3.8.13, 3.9.12, 3.10.6
- **Code we can use to reproduce**:

### Describe the problem
`read_excel` function works fine when I didn't include the parameter `header` and set it to `None` with `header=None`.
I have tested it out with an excel that contains a column of only numbers and another excel file with a column that has the 1st row as string and the rest as numbers (the 1st row should also be considered a normal row along with the rest)

### Source code / logs
**Source code**:
Case 1
```
import modin.pandas as pd

data = pd.read_excel('example.xlsx', header=None)
data
```

Case 2
```
import modin.pandas as pd

data = pd.read_excel('example_.xlsx', header=None)
data
```

**Excel file to reproduce**:
[example.xlsx](https://github.com/modin-project/modin/files/9487635/example.xlsx)
[example_2.xlsx](https://github.com/modin-project/modin/files/9487637/example_2.xlsx)

 **Log**:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/shiro/.local/share/virtualenvs/taipy-core-fdyg53sb/lib/python3.9/site-packages/modin/logging/logger_metaclass.py", line 68, in log_wrap
    return method(*args, **kwargs)
  File "/Users/shiro/.local/share/virtualenvs/taipy-core-fdyg53sb/lib/python3.9/site-packages/modin/pandas/dataframe.py", line 215, in __repr__
    result = repr(self._build_repr_df(num_rows, num_cols))
  File "/Users/shiro/.local/share/virtualenvs/taipy-core-fdyg53sb/lib/python3.9/site-packages/modin/logging/logger_metaclass.py", line 68, in log_wrap
    return method(*args, **kwargs)
  File "/Users/shiro/.local/share/virtualenvs/taipy-core-fdyg53sb/lib/python3.9/site-packages/modin/pandas/base.py", line 203, in _build_repr_df
    return self.iloc[indexer]._query_compiler.to_pandas()
  File "/Users/shiro/.local/share/virtualenvs/taipy-core-fdyg53sb/lib/python3.9/site-packages/modin/logging/logger_metaclass.py", line 68, in log_wrap
    return method(*args, **kwargs)
  File "/Users/shiro/.local/share/virtualenvs/taipy-core-fdyg53sb/lib/python3.9/site-packages/modin/core/storage_formats/pandas/query_compiler.py", line 259, in to_pandas
    return self._modin_frame.to_pandas()
  File "/Users/shiro/.local/share/virtualenvs/taipy-core-fdyg53sb/lib/python3.9/site-packages/modin/logging/logger_metaclass.py", line 68, in log_wrap
    return method(*args, **kwargs)
  File "/Users/shiro/.local/share/virtualenvs/taipy-core-fdyg53sb/lib/python3.9/site-packages/modin/core/dataframe/pandas/dataframe/dataframe.py", line 115, in run_f_on_minimally_updated_metadata
    result = f(self, *args, **kwargs)
  File "/Users/shiro/.local/share/virtualenvs/taipy-core-fdyg53sb/lib/python3.9/site-packages/modin/core/dataframe/pandas/dataframe/dataframe.py", line 2840, in to_pandas
    ErrorMessage.catch_bugs_and_request_email(
  File "/Users/shiro/.local/share/virtualenvs/taipy-core-fdyg53sb/lib/python3.9/site-packages/modin/error_message.py", line 70, in catch_bugs_and_request_email
    raise Exception(
Exception: Internal Error. Please visit https://github.com/modin-project/modin/issues to file an issue with the traceback and the command that caused this error. If you can't file a GitHub issue, please email bug_reports@modin.org.
Internal and external indices on axis 1 do not match.
>>> data = pd.read_excel('tests/data_sample/example.xlsx', header=None)
UserWarning: Parallel `read_excel` is a new feature! If you run into any problems, please visit https://github.com/modin-project/modin/issues. If you find a new issue and can't file it on GitHub, please email bug_reports@modin.org.

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4924 
- [ ] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Review by mvashishtha — DISMISSED** [review]
thank you @anmyachev !!!

File: modin/core/io/text/excel_dispatcher.py [hunk 1]
```diff
@@ -132,22 +125,38 @@ def _read(cls, io, **kwargs):
             while end_of_row_tag not in sheet_block:
                 sheet_block += f.read(EXCEL_READ_BLOCK_SIZE)
             idx_of_header_end = sheet_block.index(end_of_row_tag) + len(end_of_row_tag)
-            sheet_header = sheet_block[:idx_of_header_end]
-            # Reset the file pointer to begin at the end of the header information.
-            f.seek(idx_of_header_end)
+            sheet_header_with_first_row = sheet_block[:idx_of_header_end]
+
+            if not kwargs["header"] is None:
```
Comment (vnlitvinov) [inline modin/core/io/text/excel_dispatcher.py:130]:
```suggestion
            if kwargs["header"] is not None:
```
I personally like this form (`foo is not bar` vs `not foo is bar`) more, as I feel it's more natural to read

File: modin/core/io/text/excel_dispatcher.py [hunk 2]
```diff
@@ -132,22 +125,38 @@ def _read(cls, io, **kwargs):
...
+                # Reset the file pointer to begin at the end of the header information.
+                f.seek(idx_of_header_end)
+                sheet_header = sheet_header_with_first_row
+            else:
+                start_of_row_tag = b"<row"
+                idx_of_header_start = sheet_block.index(start_of_row_tag)
```
Comment (vnlitvinov) [inline modin/core/io/text/excel_dispatcher.py:136]: what if file is malformed, and `idx_of_header_start` is `-1` (i.e. no `<row` is found)?

Comment (vnlitvinov) [reply modin/core/io/text/excel_dispatcher.py:136]: Let's make it in a separate one.

**Review by vnlitvinov — APPROVED** [review]
LGTM!

**Comment by anmyachev** [comment]
@mvashishtha can we merge it?

**Pull request merged**